velox/functions/remote/utils/restserver/RemoteFunctionRestHandler.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+
+#include "velox/vector/VectorStream.h"
+
+namespace facebook::velox::functions {
+
+class RemoteFunctionRestHandler {
+ public:
+  virtual ~RemoteFunctionRestHandler() = default;
+
+  // Handles the actual function execution
+  virtual void handleRequest(
+      std::unique_ptr<folly::IOBuf> inputBuffer,
+      VectorSerde* serde,
+      memory::MemoryPool* pool,
+      std::function<void(folly::IOBuf&&)> sendResponse) = 0;
```

Comment:
Any particular reason this is wrapped as a callback function ? The API would be simpler if you just took at input IOBuf and returned an output IOBuf as well and let the RestService handle the communication.